### ISSUE TITLE
Providers

### ISSUE THREAD

**Issue by 0xMRTT** [Bavarder/Bavarder#17]
There is currently only
- baichat
- catgpt
- huggingchat
- openaigpt35turbo
- openaigpt4
enabled by default.

But there is 
- alpacalora
- baichat
- bard
- catgpt
- hfdialogpt
- hfgoogleflant5xxl
- hfgoogleflanu12
- hfgpt2
- hfopenassistantsft1pythia12b
- huggingchat
- openaigpt35turbo
- openaigpt4
- openaitextdavinci003
available.

The list of available provider is too big so I would like to ask which providers do you want to see by default.

### Adding a new provider 

You can edit the dconf key `enabled-providers`

#### Flatpak

``` shell
nano ~/.var/app/io.github.Bavarder.Bavarder/config/glib-2.0/settings/keyfile
```

And add a new line with 
``` shell
enabled-providers = [ 'bard', 'huggingchat' ] # add the providers you want from the previous list
```

### ROADMAP

- [ ] Add a switch in pref for enabling/disabling a provider

**Comment by 0xMRTT** [comment]
https://codeberg.org/Bavarder/Bavarder/issues/15

**Comment by darkdragon-001** [comment]
Bard would be nice.

**Comment by gargullia** [comment]
Add StarCoder* https://huggingface.co/bigcode/starcoder

*StarCoder is an alternative to services like Copilot

more info: https://www.servicenow.com/company/media/press-room/starcoder.html

**Comment by 0xMRTT** [comment]
That's a nice but code formatting needs to be done first otherwise code will look ugly

**Comment by 0xMRTT** [comment]
0.1.7 has been released :rocket: featuring a new preferences window allowing to enable/disable providers

**Comment by prog-amateur2** [comment]
Hello, did you hear about [Koala](https://koala.lmsys.org/) (the demo webpage is not always reachable) ? 
It is open source, unrestricted IA, made by [Berkeley](https://bair.berkeley.edu/blog/2023/04/03/koala/), and it seems that an [internal leaked document from a Google employee](https://www.semianalysis.com/p/google-we-have-no-moat-and-neither) mentions that it could be better than retricted AI like OpenAI and Bard. 

I think it could be really good to have a real open source alternative to ChatGPT or Bard for instance. Thank you.

**Comment by 0xMRTT** [comment]
If there is an API available somewhere, i can add it.

I tried to use HF but the model is too large

**Comment by prog-amateur2** [comment]
Thank you very much for your quick feedback. It seems that it is possible to install Koala [locally](https://github.com/young-geng/EasyLM/blob/main/docs/koala.md), but not through an API yet (or I didn't find it). I hope it will become more popular and we can have an API available, anyway, I have asked if it exists or not.

**Comment by 0xMRTT** [comment]
Nice, I prefer to use API because the app keep smaller

**Comment by CaptainMorgan12** [comment]
For all these model providers could a `!` be added next to the provider in settings or maybe a brief description of what the model is used for typically, which language it supports, technology it is based on. User would for example select the `!` and it would provide details next to the model. A different approach could be to have a small text page `Wiki` for each model listed in the app, with model title and description under each as a long list. This wouldn't have to be kept up to date necessarily, just a few extra pieces of information that allows a user to de-activate those quickly that wouldn't be useful.

**Comment by 0xMRTT** [comment]
@CaptainMorgan12 I added a popover displaying the name of the provider, the version and if set a description and the list of supported languages. 

If you want to contribute, you can add `description = "The description` and `languages = `english and german` to each provider like:

```python
class Provider(BavarderProvider):
    name = "..." # already set
    slug = "..." # already set
    description = "..."
    languages = "..."
```
Thanks!!

**Comment by 0xMRTT** [comment]
> Add StarCoder* https://huggingface.co/bigcode/starcoder
> 
> *StarCoder is an alternative to services like Copilot
> 
> more info: https://www.servicenow.com/company/media/press-room/starcoder.html

Added starcoder. Wait for the next release

**Comment by 0xMRTT** [comment]
Starcoder will be better in 0.2.0 because in 0.2.0 I added formatting

**Comment by Gandalf-the-Blue** [comment]
Hi,
Is it possible to add a custom provider i.e. a self-hosted instance of say local AI (https://github.com/go-skynet/LocalAI). It uses the OpenAI API standard
So if it is possible to change the base url and add an API key, it should work, right?

**Comment by 0xMRTT** [comment]
Yep, that can be easily doable like what I did in [imaginer](https://imaginer.codeberg.page) for custom providers

**Comment by 0xMRTT** [comment]
This has been added. The documentation is available [here](https://bavarder.codeberg.page/help/local/#custom-model)

**Comment by Gandalf-the-Blue** [comment]
> This has been added. The documentation is available [here](https://bavarder.codeberg.page/help/local/#custom-model)

Wow! Thank you! That's amazing speed :)
When do you expect the next release to be'?

**Comment by 0xMRTT** [comment]
Now :)

**Comment by 0xMRTT** [comment]
I removed it because I replaced hugging chat. But i'm planning to re-add it

**Comment by 0xMRTT** [comment]
It will be available either as API or as a local model :)

**Comment by tomoqv** [comment]
I am trying to use Bavarder with ChatGPT 3.5 Turbo, but can't for my life find which "API url" to enter in the Bavarder settings (below the API key itself). I am getting the error message "I'm having trouble connecting to the API, please check your internet connection." or "You don't have access to this model, please check your plan and billing details." depending on what I enter. I have enabled payment in OpenAI and created a project where I enabled ChatGPT 3.5 Turbo.

What do I need to enter to make it work?

**Comment by 0xMRTT** [comment]
In less than a month, I'm going to release a new version of Bavarder with a lot of bug fixes and new features like image generation. Right now, if you wanna use bavarder, you can build it from source.

**Comment by 0xMRTT** [comment]
not planned as it's now an offline first app
